### PR TITLE
refactor(npm): create general use `NpmPackageResolver`

### PR DESCRIPTION
**Pull request by dsherret**
This is preparation for a local node_modules folder resolver.

1. `DenoDirNpmResolver` -> `RequireNpmResolver`
1. Hide `GlobalNpmPackageResolver`
2. Create general use `NpmPackageResolver` that has an internal implementation that can be swapped out.

### CODE REVIEW

File: cli/npm/mod.rs [hunk 1]
```diff
@@ -3,422 +3,14 @@
...
-  ) -> Result<LocalNpmPackageInfo, AnyError> {
-    let pkg_id = self
-      .cache
-      .resolve_package_id_from_specifier(specifier, &self.registry_url)?;
-    Ok(self.local_package_info(&pkg_id))
-  }
-}
-
-#[derive(Clone, Debug)]
-pub struct NpmPackageResolverSnapshot {
```
Comment (dsherret) [inline cli/npm/mod.rs:255]: In the interest of time, I completely removed this for now and will add it back in the future when we actually go to implement lsp support.

File: cli/npm/resolvers/common.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+/// Information about the local npm package.
+pub struct LocalNpmPackageInfo {
+  /// Unique identifier.
+  pub id: NpmPackageId,
+  /// Local folder path of the npm package.
+  pub folder_path: PathBuf,
+}
+
+pub trait InnerNpmPackageResolver: Send + Sync {
```
Comment (ry) [inline cli/npm/resolvers/common.rs:24]: Why "Inner" ? Usually I wouldn't expect that name exposed from a module

Comment (dsherret) [reply cli/npm/resolvers/common.rs:24]: I'm not sure about a name. It's exposed from this module, but private to `cli/npm/resolvers` (not exported from there).

**Review by ry — APPROVED** [review]
LGTM

**Pull request merged**